packages/monaco-graphql/src/graphqlMode.ts

```diff
@@ -25,82 +28,87 @@ export function setupMode(defaults: LanguageServiceAPI): IDisposable {
   const client = new WorkerManager(defaults);
   const { languageId } = defaults;
   disposables.push(client);
-  const worker: languageFeatures.WorkerAccessor = (
-    ...uris: Uri[]
-  ): Promise<GraphQLWorker> => {
-    try {
-      return client.getLanguageServiceWorker(...uris);
-    } catch (err) {
-      throw Error('Error fetching graphql language service worker');
-    }
-  };
 
-  defaults.setWorker(worker);
-
-  monaco.languages.setLanguageConfiguration(languageId, richLanguageConfig);
+  monaco.languages.setLanguageConfiguration(languageId, conf);
   monaco.languages.setMonarchTokensProvider(languageId, monarchLanguage);
 
-  function registerFormattingProvider(): void {
-    const { modeConfiguration } = defaults;
-    if (modeConfiguration.documentFormattingEdits) {
-      providers.push(
-        monaco.languages.registerDocumentFormattingEditProvider(
-          defaults.languageId,
-          new languageFeatures.DocumentFormattingAdapter(worker),
-        ),
-      );
+  defaults.onSchemaLoaded(_api => {
```

Comment:
this is probably one of the most important fundamental improvements to `monaco-graphql` in this PR. 

previously, the language service would load whether a schema was present or not (a behaviour I had inadvertently mirrored from `monaco-json`, which actually makes sense for them), and with webworkers, there was no single source of truth to know whether each webworker had independently requested and loaded it's own schema. this would lead to console errors and unnecessary churn for client implementations.

here, we use the event emitted `onSchemaLoaded` to wait for the schema to be present before loading any schema driven language features. note that schema-agnostic features such as language configuration, the tokens provider and the formatting provider ala `prettier/standalone` are registered above as immediately as possible, without waiting for `onSchemaLoaded`. 

as an argument to the `onSchemaLoaded` EventEmitter callback, the`api` instance is passed as an argument because why not? but the existing `GraphQLAPI` instance we have above is already sufficient to use for schema driven features at this point. `getSchema()` will read from the in memory cache, and `reloadSchema()` and `setSchema()` or `setSchemaConfig()` etc will trigger a new schema loading event, which will then re-load the language features as we expect

Reply:
I believe @cshaver and others have reccomended this idea before, which makes way more sense!